src/ModelContextProtocol/Authentication/Types/AuthenticationSchemeMismatchException.cs

```diff
@@ -0,0 +1,33 @@
+namespace ModelContextProtocol.Authentication.Types;
+
+/// <summary>
+/// Exception thrown when no compatible authentication scheme can be found between the client and server.
+/// </summary>
+public class AuthenticationSchemeMismatchException : Exception
```

Comment:
Do we need a custom exception type? e.g. will a consumer need to programmatically inspect ServerSchemes and ProviderSchemes? If it's not strictly needed, it'd be preferable to just use InvalidOperationException or something like that, with a message that contains the relevant diagnostic details.